runtime/src/in_mem_accounts_index.rs

```diff
@@ -1416,6 +1416,7 @@ impl<'a> FlushGuard<'a> {
     /// Set the `flushing` atomic flag to true.  If the flag was already true, then return `None`
     /// (so as to not clear the flag erroneously).  Otherwise return `Some(FlushGuard)`.
     #[must_use = "if unused, the `flushing` flag will immediately clear"]
+    #[allow(clippy::unnecessary_lazy_evaluations)]
```

Comment:
Someone will probably inadvertently change this at some point -- any chance we can add the `allow` specifically on the offending line and / or a comment explaining why this is needed?

Reply:
Done